File: java/src/main/java/io/fivexer/sdk/model/WorkflowRouting.java
package io.fivexer.sdk.model;

/**
 * A branch rule. {@code condition} is evaluated against the step result and the first match
 * wins. Used both when saving a definition and when reading one back.
 */
public class WorkflowRouting {
    private String condition;
    private String targetStepId;

    /** Response construction (Gson). */
    public WorkflowRouting() {}

    public WorkflowRouting(String condition, String targetStepId) {
        this.condition = condition;
        this.targetStepId = targetStepId;
    }

    public String getCondition() { return condition; }
    public String getTargetStepId() { return targetStepId; }
}

File: php/src/Model/WorkflowRouting.php
<?php

declare(strict_types=1);

namespace Fivexer\SDK\Model;

/**
 * A branch rule. `condition` is evaluated against the step result and the first match wins.
 * Used both when saving a definition and when reading one back.
 */
final class WorkflowRouting
{
    public function __construct(
        public readonly string $condition,
        public readonly string $targetStepId,
    ) {
    }

    /** @return array<string, mixed> */
    public function toArray(): array
    {
        return ['condition' => $this->condition, 'targetStepId' => $this->targetStepId];
    }

    /**
     * @param array<string, mixed> $data
     * @internal
     */
    public static function fromArray(array $data): self
    {
        return new self(
            condition: (string) ($data['condition'] ?? ''),
            targetStepId: (string) ($data['targetStepId'] ?? ''),
        );
    }
}
